src/importlinter/domain/contract.py

```diff
@@ -22,17 +23,23 @@ def _populate_fields(self) -> None:
         Raises:
             InvalidContractOptions if the contract options could not be matched to the fields.
         """
+        contract_options = copy.deepcopy(self.contract_options)
+
         errors = {}
         for field_name in self.__class__._get_field_names():
             field = self.__class__._get_field(field_name)
 
             try:
-                raw_data = self.contract_options[field_name]
+                raw_data = contract_options.pop(field_name)
```

Comment:
why have you changed subscript `[]` to more costly `deepcopy()` and `pop()`?

Reply:
This is done so that once we have parsed all of the contract options we can check if there are any extra ones that were passed in (which should then trigger an error as the user has set a configuration option for the contract that doesn't exist in the contract specification).

You could potentially do this by comparing `self.contract_options.keys()` with `self.__class__._get_field_names()`, but I figured this was was a bit clearer, and the cost of the `deepcopy` of a dict of strings is negligible compared to the cost of building the import graph for a large codebase.